source/decode.c

```diff
@@ -404,48 +467,89 @@ static int s_linestate_header(struct aws_http_decoder *decoder, struct aws_byte_
 
     switch (header.name) {
         case AWS_HTTP_HEADER_CONTENT_LENGTH:
-            if (s_read_int64(header.value_data, (int64_t *)&decoder->content_length) != AWS_OP_SUCCESS) {
+            if (decoder->transfer_encoding) {
+                AWS_LOGF_ERROR(
+                    AWS_LS_HTTP_STREAM,
+                    "id=%p: Incoming headers for both content-length and transfer-encoding received. This is illegal.",
+                    decoder->logging_id);
+                return aws_raise_error(AWS_ERROR_HTTP_PARSE);
+            }
+
+            if (s_read_size(header.value_data, &decoder->content_length, 10) != AWS_OP_SUCCESS) {
+                AWS_LOGF_ERROR(
+                    AWS_LS_HTTP_STREAM,
+                    "id=%p: Incoming content-length header has invalid value.",
```

Comment:
Do you want to log what the invalid value was?

Reply:
did